invenio_communities/members/services/service.py

```diff
@@ -0,0 +1,82 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2022 Northwestern University.
+#
+# Invenio-Communities is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""Members Service."""
+
+from invenio_records_resources.services.records import RecordService
+from invenio_records_resources.services.uow import RecordCommitOp, unit_of_work
+
+
+class MemberService(RecordService):
+    """Member Service."""
+
+    def _to_id(self, data):
+        """Convert data dict to dict of primary keys."""
+        # TODO: add group
+        return {
+            "community_id": data.get("community"),
+            "user_id": data.get("user"),
+        }
+
+    @unit_of_work()
+    def create(self, identity, data, uow=None):
+        """Create member."""
+        # TODO: Check permission
+        # self.require_permission(identity, "create")
+
+        # Validate data (if there are errors, .load() raises)
+        data, errors = self.schema.load(
+            data,
+            context={"identity": identity},
+        )
+
+        # It's the components that save the actual data in the record.
+        member_record = self.record_cls.create(
+            {},
+            **self._to_id(data),
+        )
+
+        # Run components
+        self.run_components(
```

Comment:
Minor: I not sure we need components. It's mainly for when a service has multiple independent concerns, and the members service has a single responsibility. Thus, I'd rather keep it out to simplify and see if we really need it. See https://inveniordm.docs.cern.ch/develop/topics/service/#service-components

Reply:
sure